infrastructure.tf

```diff
@@ -422,3 +425,116 @@ resource "aws_iam_role_policy" "cwl-firehose-staging" {
 }
 EOF
 }
+
+////
+// CW Alarms to Slack notifications
+//
+
+variable "code_deployment_bucket" {}
+variable "kms_key_arn" {}
+
+resource "aws_iam_role" "slack_notifier" {
+  name = "CloudWatchSlackNotifier"
+  path = "/service-role/"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": {
+    "Effect": "Allow",
+    "Principal": { "Service": "lambda.amazonaws.com" },
+    "Action": "sts:AssumeRole"
+  }
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "slack_notifier_kms" {
+  name = "${aws_iam_role.slack_notifier.name}-kms"
+  role = "${aws_iam_role.slack_notifier.name}"
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "kms:Decrypt"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "slack_notifier_logs" {
+  name = "${aws_iam_role.slack_notifier.name}-logs"
+  role = "${aws_iam_role.slack_notifier.name}"
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "logs:CreateLogGroup",
+            "Resource": "arn:aws:logs:us-east-1:861229788715:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": [
+                "arn:aws:logs:us-east-1:861229788715:log-group:/aws/lambda/cloudwatch-slack-notifications:*"
```

Comment:
shouldn't the account id be interpolated here rather than checked in?

Reply:
darn, true